sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -284,6 +304,11 @@ private void createRunnerAndConsumersForPTransformRecursively(
         startFunction.run();
       }
 
+      if (hasReadPTransform) {
```

Comment:
This should not be needed. queueingClient should handle case with no read ptransforms fine.

Reply:
Done